supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesCacheHitCopyingActor.scala

```diff
@@ -1,71 +1,104 @@
-            initializationData: JesBackendInitializationData,
-            serviceRegistryActor: ActorRef): Props = {
-    Props(new JesCacheHitCopyingActor(jobDescriptor, jesConfiguration, initializationData, serviceRegistryActor))
-  }
-}
-
 case class JesCacheHitCopyingActor(override val jobDescriptor: BackendJobDescriptor,
                                    jesConfiguration: JesConfiguration,
                                    initializationData: JesBackendInitializationData,
                                    serviceRegistryActor: ActorRef) extends BackendCacheHitCopyingActor {
 
   override val configurationDescriptor = jesConfiguration.configurationDescriptor
+  private lazy val jesAttributes = jesConfiguration.jesAttributes
+  private lazy val metadataJobKey = {
+    val jobDescriptorKey: BackendJobDescriptorKey = jobDescriptor.key
+    MetadataJobKey(jobDescriptorKey.call.fullyQualifiedName, jobDescriptorKey.index, jobDescriptorKey.attempt)
+  }
 
-  override def copyCachedOutputs(cachedJobOutputs: JobOutputs): Future[BackendJobExecutionResponse] = {
-    val workflowPaths = initializationData.workflowPaths
-    val jesWorkflowPaths = workflowPaths.toJesCallPaths(jobDescriptor.key)
-    val callRootPath = jesWorkflowPaths.callRootPath
-    val callRootParent = callRootPath.getParent
-    val callRootRoot = callRootPath.getRoot
-
-    //renaming requires some tech talk
-    def renameCallSpecificFiles: List[String] = {
-      val jesFileNames = List(
-        jesWorkflowPaths.returnCodeFilename,
-        jesWorkflowPaths.stderrFilename,
-        jesWorkflowPaths.stdoutFilename,
-        jesWorkflowPaths.jesLogFilename
-      )
-      jesFileNames
+  override def copyCachedOutputs(seqOfSimpletons: Seq[WdlValueSimpleton], jobDetritusFiles: Map[String,String],
+                                 returnCode: Option[Int]): Future[BackendJobExecutionResponse] = {
+
+      val gcsFileSystem = initializationData.workflowPaths.gcsFileSystemWithUserAuth
+      val jesCallPaths = initializationData.workflowPaths.toJesCallPaths(jobDescriptor.key)
+      val destinationCallRootPath: Path = jesCallPaths.callRootPath
+      val getSourceCallRootPath = jobDetritusFiles.get(jesCallPaths.callRootPathKey) match {
+        case Some(srcPath) => gcsFileSystem.getPath(srcPath)
+        case None => val error = new RuntimeException(s"The call detritus files for source cache hit aren't found for call ${jobDescriptor.call.fullyQualifiedName}")
+      }
+
+    def cacheIfPossible = {
+      getSourceCallRootPath match {
+        case Some(path: Path) =>
+          copyFiles(path)
+          updateMetadata
+          Future(SucceededResponse(jobDescriptor.key, returnCode, createJobOutputs))
+        case error: RuntimeException => Future(FailedNonRetryableResponse(jobDescriptor.key, error, Option(-1)))
+      }
     }
 
-    val gcsFileSystem = workflowPaths.gcsFileSystemWithUserAuth
+      def copyFiles(sourceCallRootPath: Path) = {
```

Comment:
Tabulation/Spacing seems a little off in this file. This `def` should be at the same level as the previous one?